accepted/2024/support-nuget-authentication-plugins-dotnet-tools.md

```diff
@@ -0,0 +1,504 @@
+The log below shows all possible subcommands.
+
+```log
+❯ .\dotnet-codeartifact-creds.exe
+Required command was not provided.
+
+Usage:
+  dotnet-codeartifact-creds [options] [command]
+
+Options:
+  --version         Show version information
+  -?, -h, --help    Show help and usage information
+
+Commands:
+  install      Installs the AWS CodeArtifact NuGet credential provider into the NuGet plugins folder.
+  configure    Sets or Unsets a configuration
+  uninstall    Uninstalls the AWS CodeArtifact NuGet credential provider from the NuGet plugins folder.
+```
+
+NuGet plugin authors could leverage the .NET SDK to allow their customers to install, uninstall, or update the plugins more efficiently.
+
+## Explanation
+
+### Functional explanation
+
+A deployment solution for .NET is the [.NET tools](https://learn.microsoft.com/dotnet/core/tools).
+These tools provide a seamless installation and management experience for NuGet packages in the .NET ecosystem.
+The use of .NET tools as a deployment mechanism has been a recurring request from users and internal partners who need to authenticate with private repositories.
+Currently, this solution works for the Windows .NET Framework.
+However, the goal is to extend this support cross-platform for all supported .NET runtimes.
+
+The reasons why `.NET tools` were chosen as the deployment mechanism are mentioned below:
+
+- NuGet plugins are console applications. A `.NET tool` is a special NuGet package that contains a console application, which presents a natural fit.
+
+- The .NET SDK has already simplified the process for customers to develop a tool.
+All that plugin authors need to do is set the `PackAsTool` MSBuild property as shown below, and then execute `dotnet pack` to generate the package:
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net6.0</TargetFramework>
+
+    <PackAsTool>true</PackAsTool>
+    <ToolCommandName>nuget-plugin-azure-artifacts-credprovider</ToolCommandName>
+    <PackageOutputPath>./nupkg</PackageOutputPath>
+
+  </PropertyGroup>
+
+</Project>
+```
+
+- Leveraging `.NET Tools` provides a standard experience for customers to share console applications, with NuGet plugins being one such use case.
+If there is an existing mechanism that works well, is familiar to customers, and fits our use case, I question the benefit of developing a new layout specifically for NuGet plugins.
+
+- Another advantage of this established `.NET tools` approach is that plugin authors won't have to maintain separate code paths for .NET Framework and .NET Core runtimes.
+All these complexities are handled by the .NET SDK by providing a native shim upon the installation of the .NET tool.
+
+Currently, at least to my understanding, the generated `.nupkg` will include a file named `DotnetToolSettings.xml` with additional metadata such as the command name, entry point, and runner.
```

Comment:
```suggestion
The generated `.nupkg` will include a file named `DotnetToolSettings.xml` with additional metadata such as the command name, entry point, and runner.
```

That is correct, it's not a guess :D 
fwiw, you can probably just link to a description, rather than cover it here.